File: pinyougou_sellergoods_interface/src/main/java/com/robin/sellergoods/service/BrandService.java
package com.robin.sellergoods.service;

import com.robin.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

public interface BrandService {

    public List<TbBrand> findAll();

    public PageResult findPage(Integer pageNum,Integer pageSize);


    /**
     * 插入数据
     * @param brand
     */
    public void add(TbBrand brand);

    public TbBrand findOne(Long id);

    /**
     * brand是更新后的数据 一定要带有Id
     * @param brand
     */
    public void update(TbBrand brand);

    public void delete(Long[] ids);


    public PageResult findPage(TbBrand brand, Integer pageNum, Integer pageSize);
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    this.findAll = function () {
        return $http.get("/brand/findAll.do");
    }

    this.findPage = function (pageNum,pageSize) {
        return $http.get("/brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    }

    this.save = function (id,entity) {
        var method = "add";
        if(id != null)
        {
            method = "update";
        }
        return $http.post("/brand/"+method+".do",entity);
    }

    this.findOne = function (id) {
        return $http.get("/brand/findOne.do?id="+id);
    }

    this.dele = function (ids) {
        return $http.post("/brand/dele.do",ids);
    }

    this.search = function (pageNum,pageSize,searchEntity) {
        return $http.post("/brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }

})